homeassistant/components/esphome/__init__.py

```diff
@@ -461,6 +461,7 @@ def __init__(self, entry_id: str, component_key: str, key: int):
         """Unregister callbacks."""
         for remove_callback in self._remove_callbacks:
             remove_callback()
+        self._remove_callbacks = []
```

Comment:
Didn't know that `async_will_remove` behaves this way (see https://github.com/home-assistant/home-assistant/pull/19414) - this fixes the same thing as in #19414